### DISCUSSION TITLE
Please how do I declare a custom size for a modal

### DISCUSSION THREAD

**Discussion by timi-codes** [chakra-ui/chakra-ui#2676] [Q&A] [answered]
I have tried this and it doesn't seems to work. 
![image](https://user-images.githubusercontent.com/7896429/100718032-235b0280-33bb-11eb-8e45-756c0c5801c0.png)

**Comment by timi-codes** [comment, marked as answer]
This two options worked for me
js
// Option 1: local override
`<ModalContent maxW="56rem">
    {/* Content */}
</ModalContent>`

// Option 2: global override
`const style = {
  Modal: {
    sizes: {
      xl: {
        Content: { maxWidth: "56rem" },
      },
    },
  },
};`

**Comment by npetridis** [comment]
How can I extend the modal theme and add there the size we want?

**Reply by npetridis** [reply]
> How can I extend the modal theme and add there the size we want?

I managed to do it by extending the theme modal and overriding the modal sizes like this

```
const themeOverrides: Partial<ChakraTheme> = {
  styles,
  config,
  components,
  ...foundations,
};

const theme = extendTheme(
  {
    components: { Modal }
  {,
  withDefaultColorScheme({ colorScheme: 'primary' }),
);

// Modal theme file

export const Modal = {
  parts: parts.keys,
  baseStyle,
  defaultProps,
  sizes: { '2.5xl': { dialog: { maxW: '712px' } } },
};
```
this appends the `2.5xl` size to the existing modal sizes

**Reply by karol-cf** [reply]
As of 2022, this is the correct answer since Modal is a 'Multipart component' and as defined in @chakra-ui/anatomy it has a 'dialog' part, which should be modified. However, I'm using 'maxWidth' instead of 'maxW', not sure if it makes any difference.

**Reply by sgarcia-dev** [reply]
Or for anyone else visiting even later, you can just look at the theme declaration here:
https://github.com/chakra-ui/chakra-ui/blob/be24c82948b007a8f6df4652d4ff50a85cd416f9/packages/theme/src/index.ts

And anatomy package description here:
https://github.com/chakra-ui/chakra-ui/blob/be24c82948b007a8f6df4652d4ff50a85cd416f9/packages/anatomy/src/index.ts

**Comment by Creator674** [comment]
I wanted to use Modal full screen. But smhw max-width property from chakra wont let me go that easy.

So i've tried to reset 'max-width' prop to initial, but it didn't help me, chakra was still using its variable.
It turns out, that in order to rewrite this property, you need to specify both forms (maxW and maxWidth)

It is weird behaviour.

My solution:
![image](https://user-images.githubusercontent.com/119970146/208194285-5644babf-3b24-4561-9b6d-a03d61c6d152.png)
